### PR TITLE
Mapper and  service for user/task

### PR DESCRIPTION
**Pull request by mgandore**

### CODE REVIEW

File: src/task/task.mapper.ts [hunk 1]
```diff
@@ -1,4 +1,51 @@
 import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+
+import { User } from "src/user/user.entity";
+import { Repository } from "typeorm";
```
Comment (Andrei-Suta) [inline src/task/task.mapper.ts]: adauga importul pentru Repository deasupra celor din proiect

**Review by Andrei-Suta — CHANGES_REQUESTED** [review]

**Review by Andrei-Suta — APPROVED** [review]

File: src/task/task.service.ts [hunk 2]
```diff
@@ -1,4 +1,38 @@
...
+		return this.taskMapper.toDTOs(taskEntities);
+	}
+
+	public async getById(id: number): Promise<TaskDTO> {
+		const taskEntity: Task = await this.taskRepository.findOneBy({ id });
+		return this.taskMapper.toDTO(taskEntity);
+	}
+
+	public async create(task: TaskDTO): Promise<TaskDTO> {
+		const taskEntity: Task = await this.taskRepository.save(this.taskRepository.create(await this.taskMapper.toEntity(task)));
```
Comment (danielursu) [inline src/task/task.service.ts]: separate

**Review by danielursu — CHANGES_REQUESTED** [review]

File: src/task/task.service.ts [hunk 3]
```diff
@@ -26,7 +26,8 @@ export class TaskService {
 	}
 
 	public async create(task: TaskDTO): Promise<TaskDTO> {
```
Comment (danielursu) [inline src/task/task.service.ts]: taskDTO

File: src/user/user.service.ts [hunk 4]
```diff
@@ -21,7 +21,8 @@ export class UserService {
 	}
 
 	public async create(user: UserDTO): Promise<UserDTO> {
```
Comment (danielursu) [inline src/user/user.service.ts]: userDTO

File: src/task/task.service.ts [hunk 5]
```diff
@@ -26,7 +26,8 @@ export class TaskService {
...
-		const taskEntity: Task = await this.taskRepository.save(this.taskRepository.create(await this.taskMapper.toEntity(task)));
+		const taskEntity: Task = await this.taskMapper.toEntity(task);
+		this.taskRepository.save(this.taskRepository.create(taskEntity));
```
Comment (danielursu) [inline src/task/task.service.ts]: do you need await on create and save method?

Comment (danielursu) [reply src/task/task.service.ts]: try to separate create from save

File: src/user/user.service.ts [hunk 6]
```diff
@@ -21,7 +21,8 @@ export class UserService {
...
-		const userEntity: User = await this.userRepository.save(this.userRepository.create(await this.userMapper.toEntity(user)));
+		const userEntity: User = await this.userMapper.toEntity(user);
+		this.userRepository.save(this.userRepository.create(userEntity));
 		return this.userMapper.toDTO(userEntity);
```
Comment (danielursu) [inline src/user/user.service.ts]: same

**Review by danielursu — CHANGES_REQUESTED** [review]

File: src/user/user.service.ts (hunk 6 above)
Comment (mgandore) [reply src/user/user.service.ts]: oh, I got it wrong :))

**Review by danielursu — APPROVED** [review]

**Pull request merged**